app/models/concerns/stripe_payment_status.rb

```diff
@@ -0,0 +1,26 @@
+require 'active_support/concern'
+
+module StripePaymentStatus
+  extend ActiveSupport::Concern
+
+  # Returns the current payment status from a live call to the Stripe API.
+  # Returns nil if the payment is not a Stripe payment or does not have a payment intent.
+  # If the payment requires authorization the status will be "requires_action".
+  # If the payment has been captured the status will be "succeeded".
+  # Docs: https://stripe.com/docs/api/payment_intents/object#payment_intent_object-status
+  def stripe_status
```

Comment:
if these methods are only used for support/live I think that should be mentioned in the comments so that they are not considered dead code.

Reply:
I think we'll _probably_ want to start utilising (at least) the `#stripe_captured?` method elsewhere in the codebase fairly soon, and maybe both methods. We need to increase robustness in some key places where we sometimes end up with payments captured in Stripe but still marked as "pending" in OFN.

I imagine that means you'll be more adamant about moving this to a service :sweat_smile:

Reply:
> I think we'll probably want to start utilising (at least) the #stripe_captured? method elsewhere in the codebase fairly soon

I'm pretty wary of using these methods for anything but debugging, to be honest. I don't think we're that far off from having the the payment state that's in the database always match what's in Stripe, and I don't think we should have to resort to making a live call to Stripe any time we want to know if a payment is captured.

Reply:
I also think it's best to reassess the existing architecture so it's more robust rather than relying on defensive coding.